src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/HiddenFieldCheckTest.java

```diff
@@ -403,4 +411,53 @@ public void testReceiverParameter() throws Exception {
         verify(checkConfig, getPath("InputHiddenFieldReceiver.java"), expected);
     }
 
+    /**
+     * We cannot reproduce situation when visitToken is called and leaveToken is not.
+     * So, we have to use reflection to be sure that even in such situation
+     * state of the field will be cleared.
+     *
+     * @throws Exception when code tested throws exception
+     */
+    @Test
+    public void testClearState() throws Exception {
+        final HiddenFieldCheck check = new HiddenFieldCheck();
+        final Optional<DetailAST> classDef = TestUtil.findTokenInAstByPredicate(
+            JavaParser.parseFile(
+                new File(getPath("InputHiddenField.java")),
+                JavaParser.Options.WITHOUT_COMMENTS),
+            ast -> ast.getType() == TokenTypes.CLASS_DEF);
+
+        assertTrue("Ast should contain CLASS_DEF", classDef.isPresent());
+        assertTrue("State is not cleared on beginTree",
+                TestUtil.isStatefulFieldClearedDuringBeginTree(
+                        check,
+                        classDef.get(),
+                        "frame",
+                    frame -> {
```

Comment:
please move this lamda to some class with good name or ... it is very unclear code.